### DISCUSSION TITLE
Does kohya support --negative_prompt in image caption .txt files?

### DISCUSSION THREAD

**Discussion by araleza** [bmaltais/kohya_ss#1392] [Q&A]
A1111 supports image caption files (e.g.: .caption / .txt) that contain various tags such as:

   "--prompt photo of a city center --negative_prompt people"

This A1111 ability is described here, along with some other useful tags that can be placed into prompt files:

   https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/5699#issuecomment-1377612550

Will kohya read the negative prompt when training, or ignore it and treat it as another positive keyword?  It's much easier to caption images well when negative prompts can be used.

I found that 'prompt weights' are supported by kohya, but in A1111 adding (people:-1) to the positive prompt definitely does not work the same as placing that word in the negative prompt.

**Comment by Zokreb** [comment]
I second that request, I had a similar question here : #370

Although, I don't think that Bernard can do much about it. It would be more for kohya_ss actually ?

**Comment by araleza** [comment]
I've noticed that in the 'Sample prompts' section at the bottom of the Lora parameters, it says this:

> Sample prompts
> Enter one sample prompt per line to generate multiple samples per cycle. Optional specifiers include: --w (width), --h (height), --d (seed), --l (cfg scale), --s (sampler steps) and **--n (negative prompt)**. To modify sample prompts during training, edit the prompt.txt file in the samples directory.

Maybe it's supported for the training images, but with --n instead of --negative_prompt.  It's hard to tell if it really is though, without going through the code itself, or setting up a test that would show obvious differences based on whether it was working or not.

**Reply by Zokreb** [reply]
I really believe this is for sample generation only. But I admit, I've never tried to modify the captions.
